src/controllers/expenses.js

```diff
@@ -0,0 +1,118 @@
+'use strict';
+
+const expensesServices = require('../services/expenses');
+const { getUserById } = require('../services/users');
+
+async function getAll(req, res) {
+  const params = req.query;
+  const expenses = await expensesServices.getFilteredExpenses(params);
+
+  res.send(expenses.map(expensesServices.normalizeExpense));
+}
+
+async function getOne(req, res) {
+  const { expenseId } = req.params;
+
+  if (!expenseId) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const foundExpense = await expensesServices.getExpenseById(expenseId);
+
+  if (!foundExpense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.send(expensesServices.normalizeExpense(foundExpense));
+}
+
+async function addNewExpense(req, res) {
+  const {
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+  } = req.body;
+
+  const foundUser = await getUserById(userId);
+
+  const isAllDataProvided = (foundUser
+    && userId
+    && spentAt
+    && title
+    && amount
+    && category
+  );
+
+  if (!isAllDataProvided) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const expenseData = req.body;
```

Comment:
Use object destructuring for better readability when creating expenseData from req.body